crates/cadenza-eval/src/compiler.rs

```diff
@@ -32,6 +40,8 @@ pub struct Compiler {
     units: UnitRegistry,
     /// Type inferencer for lazy type checking.
     type_inferencer: TypeInferencer,
+    /// IR generator for code generation.
+    ir_generator: IrGenerator,
```

Comment:
I wonder if it would be a good idea to make this optional in the case that we just want to type check a file and not have to pay any costs for the IR/codegen.